src/storage/mvcc/reader/point_getter.rs

```diff
@@ -0,0 +1,783 @@
+use crate::storage::{CF_DEFAULT, CF_WRITE};
+
+use super::util::CheckLockResult;
+
+/// `PointGetter` factory.
+pub struct PointGetterBuilder<S: Snapshot> {
+    snapshot: S,
+    multi: bool,
+    fill_cache: bool,
+    omit_value: bool,
+    isolation_level: IsolationLevel,
+    ts: u64,
+}
+
+impl<S: Snapshot> PointGetterBuilder<S> {
+    /// Initialize a new `PointGetterBuilder`.
+    pub fn new(snapshot: S, ts: u64) -> Self {
+        Self {
+            snapshot,
+            multi: true,
+            fill_cache: true,
+            omit_value: false,
+            isolation_level: IsolationLevel::Si,
+            ts,
+        }
+    }
+
+    /// Set whether or not to get multiple keys.
+    ///
+    /// Defaults to `true`.
+    #[inline]
+    pub fn multi(mut self, multi: bool) -> Self {
+        self.multi = multi;
+        self
+    }
+
+    /// Set whether or not read operations should fill the cache.
+    ///
+    /// Defaults to `true`.
+    #[inline]
+    pub fn fill_cache(mut self, fill_cache: bool) -> Self {
+        self.fill_cache = fill_cache;
+        self
+    }
+
+    /// Set whether values of the user key should be omitted. When `omit_value` is `true`, the
+    /// length of returned value will be 0.
+    ///
+    /// Previously this option is called `key_only`.
+    ///
+    /// Defaults to `false`.
+    #[inline]
+    pub fn omit_value(mut self, omit_value: bool) -> Self {
+        self.omit_value = omit_value;
+        self
+    }
+
+    /// Set the isolation level.
+    ///
+    /// Defaults to `IsolationLevel::Si`.
+    #[inline]
+    pub fn isolation_level(mut self, isolation_level: IsolationLevel) -> Self {
+        self.isolation_level = isolation_level;
+        self
+    }
+
+    /// Build `PointGetter` from the current configuration.
+    pub fn build(self) -> Result<PointGetter<S>> {
+        // If we only want to get single value, we can use prefix seek.
+        let write_cursor = CursorBuilder::new(&self.snapshot, CF_WRITE)
+            .fill_cache(self.fill_cache)
+            .prefix_seek(!self.multi)
+            .build()?;
+
+        Ok(PointGetter {
+            snapshot: self.snapshot,
+            multi: self.multi,
+            fill_cache: self.fill_cache,
+            omit_value: self.omit_value,
+            isolation_level: self.isolation_level,
+            ts: self.ts,
+
+            statistics: Statistics::default(),
+
+            write_cursor,
+            write_cursor_valid: true,
+            lock_cursor: None,
+            lock_cursor_valid: false,
+            default_cursor: None,
+
+            drained: false,
+        })
+    }
+}
+
+/// This struct can be used to get the value of user keys. Internally, rollbacks are ignored and
+/// smaller version will be tried. If the isolation level is Si, locks will be checked first.
+///
+/// Use `PointGetterBuilder` to build `PointGetter`.
+pub struct PointGetter<S: Snapshot> {
+    snapshot: S,
+    multi: bool,
```

Comment:
how about `near_seekable`?

Reply:
I think `multi` is more understandable. `near_seekable` is more related to the implementation detail.